src/autoschedulers/anderson2021/AutoSchedule.cpp

```diff
@@ -0,0 +1,821 @@
+/*
+  This file is the core of the autoscheduler. Most of the code here is
+  about navigating the search space and computing the
+  featurization. This also contains the top-level interface into the
+  autoscheduler.
+
+  The most interesting classes to look at are:
+
+  LoopNest               Represents one node in our tree representation of loop nests.
+  State                  A state in the beam search. Holds a root loop nest.
+
+  Interesting functions below are:
+
+  generate_schedule            The top-level entrypoint, which computes and applies a schedule to a Halide pipeline
+  optimal_schedule             Runs the passes of the coarse-to-fine beam search
+  optimal_schedule_pass        Runs a single pass of beam search
+  LoopNest::compute_features   Recursively walks over a loop nest tree, computing our featurization using Halide's analysis tools.
+  LoopNest::apply              Actually apply a computed schedule to a Halide pipeline
+  State::generate_children     Generates successor states to a state in the beam search
+
+  Environment variables used (directly or indirectly):
+
+  HL_DEBUG_AUTOSCHEDULE
+  If set, is used for the debug log level for auto-schedule generation (overriding the
+  value of HL_DEBUG_CODEGEN, if any).
+
+  HL_PERMIT_FAILED_UNROLL
+  Set to 1 to tell Halide not to freak out if we try to unroll a loop that doesn't have a constant extent. Should generally not be necessary, but sometimes the autoscheduler's model for what will and will not turn into a constant during lowering is inaccurate, because Halide isn't perfect at constant-folding.
+
+#ifdef HALIDE_ALLOW_LEGACY_AUTOSCHEDULER_API
+
+  Most of the settings in this Autoscheduler are controlled by the values specified via
+  an `autoscheduler.fieldname` GeneratorParam, as listed in the Anderson2021Params struct;
+  this is the preferred way to set these.
+
+  For now, however, you can (instead) control these settings via env vars;
+  doing so requires that you compile all of Halide with HALIDE_ALLOW_LEGACY_AUTOSCHEDULER_API
+  defined. (Note that this ability is deprecated, and likely to be removed in Halide 16.)
+
+  That said, here are the (legacy) env vars you can still use when HALIDE_ALLOW_LEGACY_AUTOSCHEDULER_API
+  is defined:
+
+  HL_BEAM_SIZE
```

Comment:
FYI: since HALIDE_ALLOW_LEGACY_AUTOSCHEDULER_API is now removed in Halide `main`, most of these env vars are probably no longer actually in use (settings would be controlled by GeneratorParams instead), so this should probably get pruned (subsequent PR is fine)